src/cogent3/core/new_sequence.py

```diff
@@ -0,0 +1,2413 @@
+        """Returns copy of self as RNA."""
+        return self.to_moltype("rna")
+
+    def to_dna(self):
+        """Returns copy of self as DNA."""
+        return self.to_moltype("dna")
+
+    def strand_symmetry(self, motif_length=1):
+        """returns G-test for strand symmetry"""
+        counts = self.counts(motif_length=motif_length)
+        ssym_pairs = self.moltype.strand_symmetric_motifs(motif_length=motif_length)
+
+        motifs = []
+        obs = []
+        for plus, minus in sorted(ssym_pairs):
+            row = array([counts[plus], counts[minus]], dtype=int)
+            if row.max() == 0:  # we ignore motifs missing on both strands
+                continue
+
+            obs.append(row)
+            motifs.append(plus)
+
+        template = DictArrayTemplate(motifs, ["+", "-"])
+        obs = template.wrap(obs)
+        cat = CategoryCounts(obs)
+        return cat.G_fit()
+
+
+def _input_vals_pos_step(seqlen, start, stop, step):
+    start = 0 if start is None else start
+    if start > 0 and start >= seqlen:
+        # start beyond seq is an empty slice
+        return 0, 0, 1
+
+    stop = seqlen if stop is None else stop
+    if stop < 0 and abs(stop) >= seqlen:
+        # finished slice before we started seq!
+        return 0, 0, 1
+
+    start = max(seqlen + start, 0) if start < 0 else start
+
+    if stop > 0:
+        stop = min(seqlen, stop)
+    elif stop < 0:
+        stop += seqlen
+
+    if start >= stop:
+        start = stop = 0
+        step = 1
+
+    return start, stop, step
+
+
+def _input_vals_neg_step(seqlen, start, stop, step):
+    # Note how Python reverse slicing works
+    # we need to make sure the start and stop are both
+    # negative, for example "abcd"[-1:-5:-1] returns "dcba"
+    if start is None or start >= seqlen:  # set default
+        start = -1  # Done
+    elif start >= 0:  # convert to -ve index
+        start = start - seqlen
+    elif start < -seqlen:  # start is bounded by len(seq)
+        return 0, 0, 1
+
+    if stop is None:  # set default
+        stop = -seqlen - 1
+    elif stop >= 0:
+        stop -= seqlen
+
+    stop = max(stop, -seqlen - 1)  # stop should always be <= len(seq)
+
+    # checking for zero-length slice
+    return (0, 0, 1) if start < stop else (start, stop, step)
+
+
+class SliceRecordABC(ABC):
+    """Abstract base class for recording the history of operations to be applied
+    to some underlying data
+
+    Notes
+    -----
+    seq_len refers to a Python typing.Sequence object, e.g. array, str, list.
+    """
+
+    start: int
+    stop: int
+    step: int
+    seq_len: int
+    _offset: int
+
+    @abstractmethod
+    def _get_init_kwargs(self) -> dict:
+        """return required arguments for construction that are unique to the
+        subclass"""
+        ...
+
+    @abstractmethod
+    def copy(self): ...
+
+    @property
+    @abstractmethod
+    def _zero_slice(self): ...
+
+    @abstractmethod
+    def to_rich_dict(self): ...
```

Comment:
since we are not expecting serialisation from view, let's remove the abstract method `to_rich_dict` from the base class. 

(I would do this myself but I am trying to avoid git conflicts!!!)